### DISCUSSION TITLE
The node cpu is soaring every hour

### DISCUSSION THREAD

**Discussion by Estheriii** [emqx/emqx#10394] [Q&A] [answered]
### What happened?

with 120w connection
![image](https://user-images.githubusercontent.com/32813533/231669701-c57030d8-5a92-4375-8f3d-1e2b7a80357d.png)

with 480w connection
![image](https://user-images.githubusercontent.com/32813533/231670544-fa8a249c-881b-4cda-bcab-cf072438dfd8.png)

you can see the peek appear every hour 

### What did you expect to happen?

no peak 

### How can we reproduce it (as minimally and precisely as possible)?

use emqx-operator 2.1.1 to start a emqx cluster in k8s 1.23
cat << "EOF" | kubectl apply -n amq -f -
apiVersion: apps.emqx.io/v2alpha1
kind: EMQX
metadata:
   name: emqx
spec:
   image: emqx:5.0
   coreTemplate:
     metadata:
       name: emqx-core
       labels:
         apps.emqx.io/role: core
     spec:
       replicas: 3
       resources:
          limits:
            cpu: "8"
            memory: 20Gi
   replicantTemplate:
     metadata:
       name: emqx-repl
       labels:
         apps.emqx.io/role: repl
     spec:
       replicas: 6
       resources:
          limits:
            cpu: "12"
            memory: 32Gi
   dashboardServiceTemplate:
     spec:
       type: NodePort
       ports:
         - name: "dashboard-listeners-http-bind"
           protocol: TCP
           port: 18083
           targetPort: 18083
           nodePort: 31083
EOF
than add 1200000 connection to the cluster
![image](https://user-images.githubusercontent.com/32813533/231671562-958758e9-b97a-491b-ab9f-e7646cd54b44.png)
![image](https://user-images.githubusercontent.com/32813533/231671717-1da96e9d-535d-4d43-90b2-035649a63c6f.png)
![image](https://user-images.githubusercontent.com/32813533/231673445-8b0003de-a47b-4414-b13d-27e40f1ac2e3.png)
![image](https://user-images.githubusercontent.com/32813533/231673497-9a492df1-8c71-4683-9812-28293bf0fa33.png)

### Anything else we need to know?

peak can also be seen in core node
![image](https://user-images.githubusercontent.com/32813533/231672108-a5e9ea25-3a7e-479b-9319-19086d697f8f.png)

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
# paste output here
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Log files

warning when the peak appears
[container(9).log](https://github.com/emqx/emqx/files/11218811/container.9.log)

**Comment by ieQu1** [comment]
Hello,

Are the connections created by a load-generator? If so, I would like to take a look at the loadgen configuration. It could be that all the clients are sending PINGREQ at the same time, or act coherently in some other way.

Another reason why one could observe spikes in CPU load is periodic forced garbage collection of the processes, that is meant to free up memory, but if I remember correctly it happens every 15 minutes.

P.S. I will convert this issue to discussion, because right now there's no evidence that these spike are not caused by the clients' behavior.

**Comment by Estheriii** [comment]
[container(10).log](https://github.com/emqx/emqx/files/11227785/container.10.log)
Here is a log file from operator that may help

**Comment by Estheriii** [comment]
under the same conditions，5.0.16 don't have peak
![image](https://user-images.githubusercontent.com/32813533/231955521-fe072941-8530-41d9-bfa5-76c5a7ba7c02.png)
![image](https://user-images.githubusercontent.com/32813533/231955626-ca621ef6-ee70-4487-883f-d05252052d01.png)

**Reply by Estheriii** [reply]
@ieQu1 please check this

**Reply by ieQu1** [reply]
Hello,

Can you start https://www.emqx.io/docs/en/v5.0/admin/cli.html#observer during one of the spikes and show the results?
In the attached logs I only see mnesia overload events. It doesn't tell much, because they can be caused by many things, like clients subscribing to the topics or reconnecting, etc. Running observer command will hopefully help to narrow down the issue.

Ideally I would like to see 3-4 screenshots of observer cli TUI taken during the spike, and 3-4 screenshots during normal operation for comparison.

Also I would like to know more about the clients. Are they real production devices, or are they created by the loadgen. If latter, what are the settings for it?

**Comment by Estheriii** [comment]
[normal.docx](https://github.com/emqx/emqx/files/11246983/normal.docx)
[peak1.docx](https://github.com/emqx/emqx/files/11246984/peak1.docx)
[peak2.docx](https://github.com/emqx/emqx/files/11246985/peak2.docx)
![image](https://user-images.githubusercontent.com/32813533/232403410-7caeecec-84ed-4c63-a9a5-a9f2b4be7ce8.png)
![image](https://user-images.githubusercontent.com/32813533/232403365-a9f88fad-9d6e-4c9c-9a2a-2bae3c53c301.png)
![image](https://user-images.githubusercontent.com/32813533/232403488-f086b2ce-fa9d-4c22-ae0a-c9bfe094df0b.png)

**Reply by Estheriii** [reply]
@ieQu1 please check screenshots

**Reply by savonarola** [reply]
Hi!
Could you please provide some screenshots of the output of `emqx ctl observer` command launched during the spikes?

**Reply by ieQu1** [reply]
Hi @savonarola ,

@Estheriii provided the screenshots (see the docx documents), from there I see that emqx_cm and emqx_router_helper are doing a lot of reductions. Sometimes mnesia_tm and disk_log processes make an appearance too, but it doesn't look like there is long message queue or anything. I guess it's caused by clients reconnecting for some reason?

**Reply by Estheriii** [reply, marked as answer]
> Hi @savonarola ,
> 
> @Estheriii provided the screenshots (see the docx documents), from there I see that emqx_cm and emqx_router_helper are doing a lot of reductions. Sometimes mnesia_tm and disk_log processes make an appearance too, but it doesn't look like there is long message queue or anything. I guess it's caused by clients reconnecting for some reason?

hey! I have resolve the problem, the problem is caused by the periodic call /metrics /monitor_current interface in my benchmark.
```
SupFlags = #{
        strategy => one_for_one,
        intensity => 10,
        period => 3600
    }
```
